lib/stdlib/src/lists.erl

```diff
@@ -416,8 +416,29 @@ delete(_, []) -> [].
       A :: term(),
       B :: term().
 
-zip([X | Xs], [Y | Ys]) -> [{X, Y} | zip(Xs, Ys)];
-zip([], []) -> [].
+zip(Xs, Ys) -> zip(Xs, Ys, fail).
+
+-spec zip(List1, List2, How) -> List3 when
+      How :: 'fail' | 'trim' | {'pad', {DefaultA, DefaultB}},
+      List1 :: [A],
+      List2 :: [B],
+      List3 :: [{A | DefaultA, B | DefaultB}],
+      A :: term(),
+      B :: term(),
+      DefaultA :: term(),
+      DefaultB :: term().
+
+zip([X | Xs], [Y | Ys], How) ->
+    [{X, Y} | zip(Xs, Ys, How)];
+zip([], [], _) ->
```

Comment:
@bjorng I see that in @Maria-12648430's current implementation, it does not matter what the last (`How`) parameter is if the lists are of equal length. It may be `fail` or `trim` or `{pad, ...}`, but it may also be anything else (`foo`, `"bar"`, `123`). Do you want that changed and checked for validity?